src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -0,0 +1,274 @@
+     */
+    TRANSIENT(Modifier.TRANSIENT, true, Set.of(FIELD)),
+
+     /**
+      * The access flag {@code ACC_VOLATILE}, corresponding to the
+      * source modifier {@link Modifier#VOLATILE volatile}.
+      */
+    VOLATILE(Modifier.VOLATILE, true, Set.of(FIELD)),
+
+    /**
+     * The access flag {@code ACC_SYNCHRONIZED}, corresponding to the
+     * source modifier {@link Modifier#SYNCHRONIZED synchronized}.
+     */
+    SYNCHRONIZED(Modifier.SYNCHRONIZED, true, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_NATIVE}, corresponding to the source
+     * modifier {@link Modifier#NATIVE native}.
+     */
+    NATIVE(Modifier.NATIVE, true, Set.of(METHOD)),
+
+    /**
+     * The access flag {@code ACC_INTERFACE}.
+     * @see Class#isInterface()
+     */
+    INTERFACE(Modifier.INTERFACE, false, Set.of(TYPE)),
+
+    /**
+     * The access flag {@code ACC_STRICT}, corresponding to the source
+     * modifier {@link Modifier#STRICT strictfp}.
+     */
+    STRICT(Modifier.STRICT, true, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_BRIDGE}
+     * @see Method#isBridge()
+     */
+    BRIDGE(0x00000040, false, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_VARARGS}.
+     * @see Executable#isVarArgs()
+     */
+    VARARGS(0x00000080, false, Set.of(METHOD, CONSTRUCTOR)),
+
+    /**
+     * The access flag {@code ACC_SYNTHETIC}.
+     * @see Class#isSynthetic()
+     * @see Executable#isSynthetic()
+     */
+    SYNTHETIC(0x00001000, false,
+              Set.of(TYPE, FIELD, METHOD, CONSTRUCTOR, ElementType.MODULE, PARAMETER)),
+
+    /**
+     * The access flag {@code ACC_MANDATED}.
+     */
+    MANDATED(0x00008000, false, Set.of(ElementType.MODULE, PARAMETER)),
+
+    /**
+     * The access flag {@code ACC_ANNOTATION}.
+     * @see Class#isAnnotation()
+     */
+    ANNOTATION(0x00002000, false, Set.of(TYPE)),
+
+   /**
+    * The access flag {@code ACC_ENUM}.
+    * @see Class#isEnum()
+    */
+    ENUM(0x00004000, false, Set.of(TYPE, FIELD)),
+
+   /**
+    * The access flag {@code ACC_MODULE}.
+    */
+    MODULE(0x8000, false, Set.of(TYPE))
+    ;
+
+    // May want to override toString for a different enum constant ->
+    // name mapping.
+
+    private int mask;
+    private boolean sourceModifier;
+
+    // For now, reuse ElementType rather than defining a separate
+    // type.
+    // Intentionally using Set rather than EnumSet since EnumSet is
+    // mutable.
+    private Set<ElementType> targets;
+
+    private AccessFlag(Set<ElementType> targets) {
+        this.mask = 0x0;
+        this.sourceModifier = false;
+        this.targets = targets;
+    }
+
+    private AccessFlag(int mask, boolean sourceModifier, Set<ElementType> targets) {
+        this.mask = mask;
+        this.sourceModifier = sourceModifier;
+        this.targets = targets;
+    }
+
+    /**
+     * {@return corresponding integer mask for the flag; 0 if none}
```

Comment:
0 would not be possible, right?

Reply:
Right; I'll fix the spec in the next revision. (This was a hold over of code originally copied from the Modifier class.)